### PR TITLE
feat: allow shellcheck directives to be ctrl-clicked

### LINKED ISSUE #535: `# shellcheck disable` as a hyperlink
### Discussed in https://github.com/vscode-shellcheck/vscode-shellcheck/discussions/299

<div type='discussions-op-text'>

<sup>Originally posted by **levlist** September  3, 2021</sup>
Hi,

when I review other developer's code and I often run such cases when they ignore a given directive in their code.
For example:
`# shellcheck disable=SC2116,SC2086`

It could be a helpful feature if vscode-shellcheck could transform these directive IDs to clickable hyperlinks, which would redirect the user to the description of the SC ID.

Example of my idea:
> #shellcheck disable=[SC2116](https://github.com/koalaman/shellcheck/wiki/SC2116),[SC2086](https://github.com/koalaman/shellcheck/wiki/SC2086)

What is your opinion about it?

Br,
Viktor

</div>

### PR DESCRIPTION
**Pull request by timonwong**
Fix #299
Fix #535

Signed-off-by: Timon Wong <timon86.wang@gmail.com>

### CODE REVIEW

**Comment by timonwong** [comment]
hmm, strange the ci is failing

**Comment by felipecrs** [comment]
I halted release so we can merge this first.

**Review by felipecrs — CHANGES_REQUESTED** [review]
This is awesome!

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.ts]
+indent_size = 2
+indent_style = space
+
+[*.json]
+indent_size = 2
+indent_style = space
```
[the remark this answers is not in the document]
Comment (felipecrs) [reply .editorconfig:13]: If your VS Code isn't picking up the right settings, perhaps you are missing the prettier extension?

File: cspell.json [hunk 2]
```diff
@@ -7,6 +7,7 @@
     ".vscode-test/**",
     "dist/**",
     "out/**",
+    "*.vsix",
```
[the remark this answers is not in the document]
Comment (timonwong) [reply cspell.json:10]: @felipecrs not sure, but it did in my development env during `npm test`, so I exclude it from cspell 🤦

File: .editorconfig (hunk 1 above)
[the remark this answers is not in the document]
Comment (timonwong) [reply .editorconfig:13]: Maybe in `.vscode/settings.json`?

File: cspell.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (felipecrs) [reply cspell.json:10]: got it, it's harmless to add it anyway

File: .editorconfig (hunk 1 above)
[the remark this answers is not in the document]
Comment (felipecrs) [reply .editorconfig]: Unless you have overriden something, prettier extension auto-configures this...

[the remark this answers is not in the document]
Comment (timonwong) [reply .editorconfig]: > Unless you have overriden something, prettier extension auto-configures this...

@felipecrs make sense, maybe there is something wrong in my settings

[the remark this answers is not in the document]
Comment (felipecrs) [reply .editorconfig]: Maybe not: https://github.com/prettier/prettier-vscode/issues/1859

[the remark this answers is not in the document]
Comment (felipecrs) [reply .editorconfig]: I think the best option is to add to `.vscode/settings.json` as you did indeed.

**Comment by felipecrs** [comment]
LGTM, should we merge?

**Comment by felipecrs** [comment]
About the PR name, I just wonder if the following would be more user-friendly:

feat: allow shellcheck directives to be ctrl-clicked

**Review by felipecrs — APPROVED** [review]

**Comment by timonwong** [comment]
@felipecrs Yep, I think it's ok to ship it now :)

**Comment by felipecrs** [comment]
Hey, there is something wrong

**Comment by felipecrs** [comment]
I paused the release, I think there is a bug, I will record a video

**Comment by felipecrs** [comment]
https://user-images.githubusercontent.com/29582865/161096724-c98eb451-47c6-446f-a821-440dc50efbaa.mp4

**Comment by felipecrs** [comment]
Perhaps we should try to add at least one test to the test suite? :P

**Comment by timonwong** [comment]
@felipecrs yep :joy:, the problem is that the matches contains "\n" at start which breaks rule id regex.

**Comment by timonwong** [comment]
I'll file a quick fix for that

**Pull request merged**